streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -300,11 +300,17 @@ private void removeStreamsSpecificConfigs(Map<String, Object> props) {
     }
 
     public Serde keySerde() {
-        return getConfiguredInstance(StreamsConfig.KEY_SERDE_CLASS_CONFIG, Serde.class);
+        Serde<?> serde = getConfiguredInstance(StreamsConfig.KEY_SERDE_CLASS_CONFIG, Serde.class);
+        serde.configure(originals(), true);
```

Comment:
`values()` get the parsed values based on the config definition, while users may define any key-value pairs they want for serde configurations?